### DISCUSSION TITLE
get_binlog_stream, detect if a col is a primary key

### DISCUSSION THREAD

**Discussion by timglabisch** [blackbeam/rust-mysql-simple#335] [Q&A] [answered]
Hello,

i am using `get_binlog_stream` and it works very well.
for example i get the `EventData::TableMapEvent` and the `EventData::RowsEvent`.

now i want to know the primary keys of the `RowsEvent`.

i expected something like this to work:

```rust
 EventData::RowsEvent(row_event) => {
    let table = /* .... */;
    for row in row_event.rows(table) {
        match row.unwrap() {
            (Some(before), Some(after)) => {
  
                for col in before.columns().iter() {
                    let is_primary_key = col.flags().contains(ColumnFlags::PRI_KEY_FLAG);
                    println!("{} is pk: {}", col.name_str(), is_primary_key);
                }
            },
            _ => unreachable!()
        };
    }
  }
```

the problem here is that the primary key is not detected. i also tryed to read it from `table.iter_optional_meta()` without luck.

what is the way to detect the primary key?

thanks a lot.

**Comment by blackbeam** [comment, marked as answer]
Hi. At the moment this info is not collected during BinlogRow deserialization. I think it clould be easily added into `impl MyDeserialize for BinlogRow` somewhere near to [other flags deserialization](https://docs.rs/mysql_common/latest/src/mysql_common/binlog/row.rs.html#169-175).

For now, I believe, you can read it from `TableMapEvent` using `iter_optional_meta` – variants of interest are `SimplePrimaryKey` and  `PrimaryKeyWithPrefix`. Description is [here](https://github.com/mysql/mysql-server/blob/8.0/libbinlogevents/include/rows_event.h#L481-L498)
